libcaf_test/caf/test/runner.hpp

```diff
@@ -5,8 +5,12 @@
 #pragma once
 
 #include "caf/detail/test_export.hpp"
+#include "caf/settings.hpp"
 #include "caf/test/registry.hpp"
 
+#include <optional>
+#include <regex>
```

Comment:
I didn't spot it last time. Given that `to_regex` is private static and since we never call it from this header, the member function could as well be implemented in `runner.cpp` as a free function in the anonymous namespace. That would avoid pulling in `<regex>` here for better header hygiene.

```suggestion
```